papermill/translators.py

```diff
@@ -272,13 +272,55 @@ def codify(cls, parameters):
         return content
 
 
+class CSharpTranslator(Translator) :
+
+    @classmethod
+    def translate_none(cls, val) :
+        # Can't figure out how to do this as nullable
+        raise NotImplementedError("Option type not implemented for C#.")
+
+    @classmethod
+    def translate_bool(cls, val) :
+        return 'true' if val else 'false'
+
+    @classmethod
+    def translate_int(cls, val):
+        strval = cls.translate_raw_str(val)
+        return strval + "L" if (val > 2147483647 or val < -2147483648) else strval
+
+    @classmethod
+    def translate_dict(cls, val):
+        """Translate dicts to nontyped dictionary"""
+
+        kvps = ', '.join(
+            ["new KeyValuePair<Object,Object>({} , {})".format(cls.translate_str(k), cls.translate(v))
+             for k, v in val.items()]
+        )
+        return 'new Dictionary<Object,Object?>(new [] {{ {} }})'.format(kvps)
```

Comment:
I think you could make this `Dictionary<string,Object?>` safely to help with key typing at least.